src/spikeinterface/preprocessing/tests/test_zero_padding.py

```diff
@@ -34,5 +36,119 @@ def test_zero_padding():
+    last_frame_of_original_trace = num_samples + padding_left
+
+    # Then from padding_left to the number of samples plus padding_left it should be the original traces
+    original_traces = recording.get_traces()
+    original_traces_from_padding = padded_traces[padding_left:last_frame_of_original_trace, :]
+    assert np.allclose(original_traces_from_padding, original_traces)
+
+    # After the original trace is over it should have zeros until the end of the padding right
+    assert np.allclose(padded_traces[last_frame_of_original_trace:, :], np.zeros((padding_right, num_channels)))
+
+
+@pytest.mark.parametrize("padding_left, padding_right", [(5, 5), (0, 5), (5, 0), (0, 0)])
+def test_trace_padded_recording_retrieve_original_trace(recording, padding_left, padding_right):
+    num_samples = recording.get_num_samples()
+
+    padded_recording = TracePaddedRecording(
+        parent_recording=recording,
+        padding_left=padding_left,
+        padding_right=padding_right,
+    )
+
+    # These are the limits of the original trace
+    start_frame = padding_left
+    end_frame = num_samples + padding_left
+    padded_traces = padded_recording.get_traces(start_frame=start_frame, end_frame=end_frame)
+
+    original_traces = recording.get_traces()
+    assert np.allclose(padded_traces, original_traces)
+
+
+@pytest.mark.parametrize("padding_left, padding_right", [(5, 5), (0, 5), (5, 0), (0, 0)])
+def test_trace_padded_recording_retrieve_partial_original_trace(recording, padding_left, padding_right):
+    num_samples = recording.get_num_samples()
+
+    padded_recording = TracePaddedRecording(
+        parent_recording=recording,
+        padding_left=padding_left,
+        padding_right=padding_right,
+    )
+
+    # These are the limits of the original trace
+    start_frame_original_traces = 2
+    end_frame_original_traces = num_samples - 2
+
+    start_frame = padding_left + start_frame_original_traces
+    end_frame = padding_left + end_frame_original_traces
+    padded_traces = padded_recording.get_traces(start_frame=start_frame, end_frame=end_frame)
+    original_traces = recording.get_traces(start_frame=start_frame_original_traces, end_frame=end_frame_original_traces)
+
+    assert np.allclose(padded_traces, original_traces)
+
+
+@pytest.mark.parametrize("padding_left, padding_right", [(5, 5), (0, 5), (5, 0), (0, 0)])
+def test_trace_padded_recording_retrieve_traces_with_partial_padding(recording, padding_left, padding_right):
+    num_samples = recording.get_num_samples()
+    num_channels = recording.get_num_channels()
+
+    padded_recording = TracePaddedRecording(
+        parent_recording=recording,
+        padding_left=padding_left,
+        padding_right=padding_right,
+    )
+
+    # Extract the traces with partial padding in the left and to the right
+    number_of_padded_frames_at_start = 2
+    start_frame = padding_left - number_of_padded_frames_at_start
```

Comment:
What if this is negative? (2.g. `0 - 2`)?
```suggestion
    start_frame = max(padding_left - number_of_padded_frames_at_start, 0)
```

Reply:
Hey @alejoe91 . Thanks for checking this, it does not have sense to have `number_of_padded_frames_at_start` larger than 0 in the case where `padding_start` is 0 (which was on the parametric test). I changed the test to account for this.

Somehow though, there was a `(-1 ) x (-1 ) =  1` situation because the tests passed when I parametrized,